tchannel/tornado/connection.py

```diff
@@ -122,7 +122,13 @@ def __init__(self, connection, tchannel=None, direction=None):
         self._messages = queues.Queue()
 
         # Map from message ID to futures for responses of outgoing calls.
-        self._outstanding = {}
+        self._outbound_pending_call = {}
+
+        # Map from message ID to outgoing request that is being sent.
+        self._outbound_pending_req = {}
+
+        # Map from message ID to outgoing response that is being sent.
+        self._outbound_pending_res = {}
```

Comment:
It appears that outbound pending req and res don't actually need references to the actual request and response objects. They can be sets of request IDs, or even just integers maintaining count on the number of ongoing requests and responses. @junchaowu can you confirm this?